### DISCUSSION TITLE
Distributed Arras/Moonray on Xeon Phi cluster?

### DISCUSSION THREAD

**Discussion by ryanrussell** [dreamworksanimation/openmoonray#25] [Ideas]
While skimming code, I couldn't help but notice some references to xeon phi.

```bash
moonray/scene_rdl2/lib/common/math/simd.h:/* include SSE emulation for Xeon Phi */
arras/distributed/arras4_node/cmd/node/HardwareFeatures.h:    /// Returns how many MIC (Intel's Xeon Phi) co-processors are currently present on the system
```

Not sure if the current codebase could be packaged to run on thos MICs, but this is a really interesting application if so...

Three questions:
1) Is anyone from the engineering team from dreamworks able to share whether the codebase can still be run on xeon phi's?

2) (If yes), are there any kind of benchmarks(real question: are they so slow that it's not worth the electricity?)

3) (If yes), Are there any guides that could be cobbled together around this? 

Would love to chat if anyone has devops knowledge around this :)

**Comment by kjeffery** [comment]
Sadly, the work on xeon phi has been stagnant for years. While most of the machines at DreamWorks support AVX512 within the CPU architecture (independent of xeon phi), we have not tried even that in quite some time. MoonRay does not make run-time SIMD decisions (yet), and we compile explicitly for AVX2 support. I would be very interested to see if somebody can build MoonRay with AVX512 support.

For clarification, Embree does dynamic SIMD decisions, and ISPC allows compilation with multiple SIMD architectures. We do not have that ISPC feature enabled.

**Reply by masahi** [reply]
> we compile explicitly for AVX2 support

Looking at the ISPC compile options ([like this one](https://github.com/dreamworksanimation/moonray/blob/8c0a066e866dcf2dc59ea92cd903fe6cdf8fc390/lib/rendering/shading/ispc/CMakeLists.txt#L55)), it seems moonrary is always generating AVX1 code for ISPC. By "we compile explicitly for AVX2 support", I assume Embree is using AVX2?

I have an AVX512-capable CPU (zen4) and wonder how much it can improve performance. Maybe I'll go ahead and generate AVX512 for ISPC and also make sure embree is using AVX512 as well.

**Reply by kjeffery** [reply]
You found a bug in our build system! Thank you for bringing it to our attention. We have filed an internal ticket to change to AVX2 for ISPC.
